hs-abci-examples/nameservice/test/Nameservice/Test/E2ESpec.hs

```diff
@@ -118,8 +118,7 @@ spec = do
         -- try to set a name without being the owner
         let msg = TypedMessage "SetName" (encode $ SetName satoshi addr2 "goodbye to a world")
             rawTx = mkSignedRawTransactionWithRoute "nameservice" privateKey2 msg
-        checkResp <- getCheckTxResponse rawTx
-        ensureCheckResponseCode checkResp 2
+        ensureCheckAndDeliverResponseCodes (0,2) rawTx
```

Comment:
I guess this translates to: _message is well formed but (in this context) not semantically._

Reply:
yes, it means that the tx will make it past the mempool when it passes check tx when the tx is run in a block it will produce an error code 2